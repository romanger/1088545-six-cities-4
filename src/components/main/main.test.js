import React from "react";
import renderer from "react-test-renderer";
import Main from "./main";

const places = [
  {
    id: 1,
    city: `Amsterdam`,
    type: `hotel`,
    pictures: {
      featured: `img/apartment-02.jpg`,
      gallery: [
        `img/apartment-03.jpg`,
        `img/room.jpg`,
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
      ]
    },
    title: `Beautiful & luxurious apartment at great location`,
    description: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`, `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`],
    price: 120,
    rate: 4.5,
    isPremium: true,
    isFeatured: false,
    badrooms: 3,
    maxGuests: 4,
    includes: [`Wifi`, `Heating`, `Kitchen`, `Cable TV`],
    host: {
      image: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      isSuper: true,
    }
  },
  {
    id: 2,
    city: `Amsterdam`,
    type: `room`,
    pictures: {
      featured: `img/apartment-01.jpg`,
      gallery: [
        `img/apartment-03.jpg`,
        `img/room.jpg`,
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
        `img/apartment-01.jpg`,
      ]
    },
    title: `Wood and stone place`,
    description: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`, `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`],
    price: 80,
    rate: 4.3,
    isPremium: false,
    isFeatured: true,
    badrooms: 1,
    maxGuests: 2,
    includes: [`Wifi`, `Heating`],
    host: {
      image: `img/avatar-angelina.jpg`,
      name: `Greta`,
      isSuper: false,
    }
  },
  {
    id: 3,
    city: `Amsterdam`,
    type: `house`,
    pictures: {
      featured: `img/apartment-03.jpg`,
      gallery: [
        `img/apartment-03.jpg`,
        `img/room.jpg`,
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
        `img/apartment-01.jpg`,
      ]
    },
    title: `Canal View Prinsengracht`,
    description: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`, `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`],
    price: 132,
    rate: 3.3,
    isPremium: false,
    isFeatured: false,
    badrooms: 2,
    maxGuests: 1,
    includes: [`Kitchen`, `Cable TV`],
    host: {
      image: `img/avatar-angelina.jpg`,
      name: `Sam`,
      isSuper: true,
    }
  },
  {
    id: 4,
    city: `Amsterdam`,
    type: `apartment`,
    pictures: {
      featured: `img/room.jpg`,
      gallery: [
        `img/apartment-03.jpg`,
        `img/room.jpg`,
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
        `img/apartment-01.jpg`,
      ]
    },
    title: `Nice, cozy, warm big bed apartment`,
    description: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`, `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`],
    price: 180,
    rate: 4.6,
    isPremium: true,
    isFeatured: false,
    badrooms: 3,
    maxGuests: 4,
    includes: [`Wifi`, `Heating`, `Kitchen`],
    host: {
      image: `img/avatar-angelina.jpg`,
      name: `Yosi`,
      isSuper: false,
    }
  },
];

it(`Render Main`, () => {
  const tree = renderer.create(
      <Main
        placesCount={50}
        places={places}
        onTitleClick={() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
