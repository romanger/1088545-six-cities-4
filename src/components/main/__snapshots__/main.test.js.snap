// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Main 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item"
              href="#"
            >
              <span>
                Paris
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item"
              href="#"
            >
              <span>
                Cologne
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item"
              href="#"
            >
              <span>
                Brussels
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item tabs__item--active"
            >
              <span>
                Amsterdam
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item"
              href="#"
            >
              <span>
                Hamburg
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item"
              href="#"
            >
              <span>
                Dusseldorf
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
          style={
            Object {
              "flexShrink": 0,
            }
          }
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            50
             places to stay in Amsterdam
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by
            </span>
            <span
              className="places__sorting-type"
              tabIndex="0"
            >
              Popular
              <svg
                className="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  xlinkHref="#icon-arrow-select"
                />
              </svg>
            </span>
          </form>
          <div
            className="cities__places-list places__list tabs__content"
          >
            <article
              className="cities__place-card place-card"
              onMouseOver={[Function]}
            >
              <div
                className="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="cities__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/apartment-02.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      120
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    Beautiful & luxurious apartment at great location
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Hotel
                </p>
              </div>
            </article>
            <article
              className="cities__place-card place-card"
              onMouseOver={[Function]}
            >
              <div
                className="cities__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/apartment-01.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      80
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    Wood and stone place
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Private Room
                </p>
              </div>
            </article>
            <article
              className="cities__place-card place-card"
              onMouseOver={[Function]}
            >
              <div
                className="cities__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/apartment-03.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      132
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "60%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    Canal View Prinsengracht
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  House
                </p>
              </div>
            </article>
            <article
              className="cities__place-card place-card"
              onMouseOver={[Function]}
            >
              <div
                className="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="cities__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/room.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      180
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    Nice, cozy, warm big bed apartment
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Apartment
                </p>
              </div>
            </article>
          </div>
        </section>
        <Map
          city={
            Array [
              52.38333,
              4.9,
            ]
          }
          places={
            Array [
              Object {
                "badrooms": 3,
                "city": "Amsterdam",
                "coordinates": Array [
                  52.3909553943508,
                  4.85309666406198,
                ],
                "description": Array [
                  "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                  "An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.",
                ],
                "host": Object {
                  "image": "img/avatar-angelina.jpg",
                  "isSuper": true,
                  "name": "Angelina",
                },
                "id": 1,
                "includes": Array [
                  "Wifi",
                  "Heating",
                  "Kitchen",
                  "Cable TV",
                ],
                "isFeatured": false,
                "isPremium": true,
                "maxGuests": 4,
                "pictures": Object {
                  "featured": "img/apartment-02.jpg",
                  "gallery": Array [
                    "img/apartment-03.jpg",
                    "img/room.jpg",
                    "img/apartment-01.jpg",
                    "img/apartment-02.jpg",
                    "img/apartment-01.jpg",
                    "img/apartment-02.jpg",
                  ],
                },
                "price": 120,
                "rate": 4.5,
                "title": "Beautiful & luxurious apartment at great location",
                "type": "hotel",
              },
              Object {
                "badrooms": 1,
                "city": "Amsterdam",
                "coordinates": Array [
                  52.3909553943508,
                  4.85309666406198,
                ],
                "description": Array [
                  "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                  "An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.",
                ],
                "host": Object {
                  "image": "img/avatar-angelina.jpg",
                  "isSuper": false,
                  "name": "Greta",
                },
                "id": 2,
                "includes": Array [
                  "Wifi",
                  "Heating",
                ],
                "isFeatured": true,
                "isPremium": false,
                "maxGuests": 2,
                "pictures": Object {
                  "featured": "img/apartment-01.jpg",
                  "gallery": Array [
                    "img/apartment-03.jpg",
                    "img/room.jpg",
                    "img/apartment-01.jpg",
                    "img/apartment-02.jpg",
                    "img/apartment-01.jpg",
                  ],
                },
                "price": 80,
                "rate": 4.3,
                "title": "Wood and stone place",
                "type": "room",
              },
              Object {
                "badrooms": 2,
                "city": "Amsterdam",
                "coordinates": Array [
                  52.3909553943508,
                  4.85309666406198,
                ],
                "description": Array [
                  "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                  "An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.",
                ],
                "host": Object {
                  "image": "img/avatar-angelina.jpg",
                  "isSuper": true,
                  "name": "Sam",
                },
                "id": 3,
                "includes": Array [
                  "Kitchen",
                  "Cable TV",
                ],
                "isFeatured": false,
                "isPremium": false,
                "maxGuests": 1,
                "pictures": Object {
                  "featured": "img/apartment-03.jpg",
                  "gallery": Array [
                    "img/apartment-03.jpg",
                    "img/room.jpg",
                    "img/apartment-01.jpg",
                    "img/apartment-02.jpg",
                    "img/apartment-01.jpg",
                  ],
                },
                "price": 132,
                "rate": 3.3,
                "title": "Canal View Prinsengracht",
                "type": "house",
              },
              Object {
                "badrooms": 3,
                "city": "Amsterdam",
                "coordinates": Array [
                  52.3909553943508,
                  4.85309666406198,
                ],
                "description": Array [
                  "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                  "An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.",
                ],
                "host": Object {
                  "image": "img/avatar-angelina.jpg",
                  "isSuper": false,
                  "name": "Yosi",
                },
                "id": 4,
                "includes": Array [
                  "Wifi",
                  "Heating",
                  "Kitchen",
                ],
                "isFeatured": false,
                "isPremium": true,
                "maxGuests": 4,
                "pictures": Object {
                  "featured": "img/room.jpg",
                  "gallery": Array [
                    "img/apartment-03.jpg",
                    "img/room.jpg",
                    "img/apartment-01.jpg",
                    "img/apartment-02.jpg",
                    "img/apartment-01.jpg",
                  ],
                },
                "price": 180,
                "rate": 4.6,
                "title": "Nice, cozy, warm big bed apartment",
                "type": "apartment",
              },
            ]
          }
        />
      </div>
    </div>
  </main>
</div>
`;
